packages/@azure/eventhubs/client/lib/connectionContext.ts

```diff
@@ -116,6 +144,10 @@ export namespace ConnectionContext {
         numSenders: Object.keys(connectionContext.senders).length,
         numReceivers: Object.keys(connectionContext.receivers).length
       };
+
+      // Remove all the senders link, receivers link, cbs session and management session for this AMQP connection.
+       remove(connectionContext);
```

Comment:
There is a function called `removeSession()` on the connection. How about if we use that instead?